web/src/components/pages/dashboard/challenge/weekly-challenge.tsx

```diff
@@ -1,7 +1,6 @@
 import * as React from 'react';
-import * as moment from 'moment';
 import WeeklyChallengeBoard from './weekly-challenge-board';
-import { useState, useEffect } from 'react';
+import { WeeklyChallenge } from '../../../../../../common/challenge';
```

Comment:
We need to import this as a module, not a relative file!

```
import { WeeklyChallenge } from 'common/challenge';
```

Reply:
This is the only important comment, the rest are just little notes.

Reply:
Oh, this one is indeed critical